src/lists/list.jsx

```diff
@@ -92,35 +95,32 @@ const List = React.createClass({
       ...other,
     } = this.props;
 
+    let hasSubheader = false;
+
+    if (subheader) {
+      hasSubheader = true;
+    } else {
+      const firstChild = React.Children.toArray(children)[0];
+      if (React.isValidElement(firstChild) && firstChild.type === Subheader) {
```

Comment:
Interesting, I didn't realize `child.type === Subheader` worked. Is there a reason why we're not using this patter over using `displayName` everywhere else in the codebase?